tapdb/universe.go

```diff
@@ -292,6 +295,13 @@ func (b *BaseUniverseTree) RegisterIssuance(ctx context.Context,
 		return nil, err
 	}
 
+	// Up to this point many writers can perform all required read
+	// operations to prepare and validate the keys. But since we're now
+	// actually starting to write, we want to limit this to a single writer
+	// at a time.
+	b.registrationMtx.Lock()
```

Comment:
I think we should revert this change. We want to make sure that our db concurrency control can handle situations like this. For sqlite, there's only a single writer. For postgres, the new serialization mode should detect conflicts, then abort so it can retry safely.

Reply:
Gotcha, perhaps we need to increase either the back off timeout, or the number of attempts (I just kinda chose 10 randomly). Wanting to try to tune these params, as otherwise we may end up hitting this in other areas that have more involved db transactions.